lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -552,34 +552,18 @@ WasmBytecodeGenerator::EmitBlock()
     // TODO: this needs more work. must get temp that brs can store to as a target, and do type checking
     m_labels->Push(blockLabel);
     EmitInfo blockInfo;
-    if (m_reader->IsBinaryReader())
+    EmitInfo tmpInfo;
+    bool nonEmpty = false;
+    EnterEvalStackScope();
+    while ((op = m_reader.ReadFromBlock()) != wnEND && op != wnELSE)
     {
-        EmitInfo tmpInfo;
-        bool nonEmpty = false;
-        EnterEvalStackScope();
-        while ((op = m_reader->ReadFromBlock()) != wnEND && op != wnELSE)
-        {
-            nonEmpty = true;
-            tmpInfo = EmitExpr(op);
-        }
-        ExitEvalStackScope();
-        // block yields last value
-        if (nonEmpty) blockInfo = tmpInfo;
-    }
-    else
-    {
-        op = m_reader->ReadFromBlock();
-        if (op == wnLIMIT)
-        {
-            throw WasmCompilationException(_u("Block must have at least one expression"));
-        }
-        do
-        {
-            EmitInfo info = EmitExpr(op);
-            ReleaseLocation(&info);
-            op = m_reader->ReadFromBlock();
-        } while (op != wnLIMIT);
+        nonEmpty = true;
+        tmpInfo = EmitExpr(op);
     }
+    ExitEvalStackScope();
+    // block yields last value
+    if (nonEmpty) blockInfo = tmpInfo;
```

Comment:
Well it is used to yield the last value of the block, but it could be refactored to simply use blockInfo directly and an empty block wouldn't get an assignment